src/core.ts

```diff
@@ -25,12 +25,18 @@ export interface CollationSpec {
   readonly backwards?: boolean // unsupported
 }
 
+/**
+ * Custom function to hash values to improve faster comparaisons
+ */
+export type CustomHashFunction =  (v: string) => string | number
+
 /**
  * Generic options interface passed down to all operators
  */
 export interface Options {
   readonly idKey: string;
-  readonly collation?: CollationSpec
+  readonly collation?: CollationSpec;
+  readonly customHashFunction?:CustomHashFunction
```

Comment:
similar to above, the field could be `hashFunction`

Reply:
renamed to hashFunction with type HashFunction